src/Microsoft.AspNet.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -62,6 +95,7 @@ public MemoryPoolIterator2 ProducingStart()
 
         public void ProducingComplete(MemoryPoolIterator2 end)
         {
+            // sync writes with is immedate == true as over Stream
```

Comment:
I don't think it needs to return a Task. Nothing that calls `ProducingComplete` cares about waiting for the actual write to complete. There is also no error handling logic if `ProducingComplete` fails asynchronously.

You're right that the `immediate` parameter is ignored by `StreamSocketOutput`. All writes are treated as `immediate: true`. If we wanted to change that we would need to maintain a buffer in this class internally that would only get flushed for `immediate: true` writes. I haven't done this yet, since it has no functional impact, but it could be a nice way to get a perf gain in the future.